src/cli/add.rs

```diff
@@ -268,45 +279,98 @@ pub async fn add_conda_specs_to_project(
     // Determine the best version per platform
     let mut package_versions = HashMap::<PackageName, HashSet<Version>>::new();
 
-    let platforms = if specs_platforms.is_empty() {
-        Either::Left(project.platforms().into_iter())
-    } else {
-        Either::Right(specs_platforms.iter().copied())
-    };
+    // Get the grouped environments that contain the feature
+    let grouped_environments: Vec<GroupedEnvironment> = project
+        .grouped_environments()
+        .iter()
+        .filter(|env| {
+            env.features()
+                .map(|feat| &feat.name)
+                .contains(&feature_name)
+        })
+        .cloned()
```

Comment:
I think you also have to make sure these are unique! (You can just add `.unique()` probably).